pyOCD/gdbserver/gdbserver.py

```diff
@@ -436,12 +456,38 @@ def getMemory(self, data):
             logging.debug("getMemory failed at 0x%x" % addr)
             val = 'E01' #EPERM
         return self.createRSPPacket(val)
-    
+
+    def writeMemoryHex(self, data):
+        split = data.split(',')
+        addr = int(split[0], 16)
+
+        split = split[1].split(':')
+        length = int(split[0], 16)
+
+        split = split[1].split('#')
+        data = hexStringToIntList(split[0])
+
+        if LOG_MEM:
+            logging.debug("GDB writeMemHex: addr=%x len=%x", addr, length)
+
+        try:
+            if length > 0:
+                self.target.writeBlockMemoryUnaligned8(addr, data)
```

Comment:
With deferred transfers enabled you'll also want to call flush to make sure data gets written out immediately (and any exceptions get thrown while you are still in the try block)